src/redux/reducers/Favorites/favorites.js

```diff
@@ -0,0 +1,29 @@
+import favoriteActions from '../../actions/favorites';
+import { loadFavorites, saveFavorites } from '../../store/localStorage';
+
+const initialState = loadFavorites();
+
+const favoritesReducer = (state = initialState, action) => {
+  const { type, payload } = action;
+  const newState = { ...state };
+
+  switch (type) {
+    case favoriteActions.types.ADD_FAVORITE: {
+      newState[payload.name] = { ...payload };
+      saveFavorites(newState);
```

Comment:
Since `saveFavorites` is writing to `localStorage` (a side effect), this reducer is not pure. Remember that all reducer must be pure. You can isolate this in a Saga or middleware, for example.